modules/alookup/alookup.go

```diff
@@ -43,6 +44,18 @@ func (s *Lookup) DoLookup(name, nameServer string) (interface{}, zdns.Trace, zdn
 	return s.DoTargetedLookup(name, nameServer)
 }
 
+// Verify that A record is indeed IPv4 and AAAA is IPv6
+func verifyAddress(ansType string, ip string) bool {
+	isIpv4 := net.ParseIP(ip).To4() != nil
+	isIpv6 := net.ParseIP(ip).To16() != nil
```

Comment:
`To16()` doesn't make any guarantees the address is IPv6; it will return non-nil for both IPv4 and IPv6, it merely converts either IPv4 or IPv6 into 16-byte representation:

https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/net/ip.go;l=226

if `net.ParseIP(ip).To16()` returns non-nil, you can then check for `':'` in `ip` (the string) to verify it is IPv6

Reply:
Done. Thanks!